File: clients/java/openfeature-provider/src/main/java/io/juspay/superposition/provider/ExperimentData.java
package io.juspay.superposition.provider;

import java.time.Instant;
import java.util.List;
import java.util.Map;

/**
 * Data class representing Superposition experiment data.
 */
public class ExperimentData {
    private final List<Map<String, Object>> experiments;
    private final List<Map<String, Object>> experimentGroups;
    private final Instant fetchedAt;

    public ExperimentData(
        List<Map<String, Object>> experiments,
        List<Map<String, Object>> experimentGroups
    ) {
        this.experiments = experiments;
        this.experimentGroups = experimentGroups;
        this.fetchedAt = Instant.now();
    }

    public List<Map<String, Object>> getExperiments() {
        return experiments;
    }

    public List<Map<String, Object>> getExperimentGroups() {
        return experimentGroups;
    }

    public Instant getFetchedAt() {
        return fetchedAt;
    }
}

File: clients/java/openfeature-provider/src/main/kotlin/io/juspay/superposition/openfeature/data_source/ExperimentData.kt
package io.juspay.superposition.openfeature.data_source

import uniffi.superposition_client.ExperimentConfig
import java.time.Instant

/**
 * Holds active experiments and experiment groups along with the timestamp they were fetched.
 *
 * @property data The experiment configuration data.
 * @property fetchedAt The timestamp when this data was last modified at its source.
 */
data class ExperimentData(val data: ExperimentConfig, val fetchedAt: Instant) {

    override fun toString(): String =
        "ExperimentData(data: ${data.experiments.size} experiments, " +
            "${data.experimentGroups.size} experiment groups" +
            ", fetched_at: $fetchedAt)"
}
